pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -268,6 +272,34 @@ func (c *InstallerController) manageInstallationPods(operatorSpec *operatorv1.Op
 	return false, nil
 }
 
+func (c *InstallerController) updateRevisionStatus() error {
```

Comment:
this method is ordered wrong.  You use the status to find the configmaps that need to be updated, not the other way around


```
failedvisions
currentrevisions
for each nodestatus
  add lastfailedrevision and currentrevision to list

if all currentrevisions point to the same revision, then mark it successful
each failed revisions should be marked as failed

any revision without a configmap should be skipped.
```